src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -820,6 +835,112 @@ CoordinatedRemoteTransactionsAbort(void)
 
 
 /*
+ * CoordinatedRemoteTransactionsSavepointBegin sends the SAVEPOINT command for
+ * the given sub-transaction id to all connections participating in the current
+ * transaction.
+ */
+void
+CoordinatedRemoteTransactionsSavepointBegin(SubTransactionId subId)
+{
+	dlist_iter iter;
+
+	dlist_foreach(iter, &InProgressTransactions)
+	{
+		MultiConnection *connection = dlist_container(MultiConnection, transactionNode,
+													  iter.cur);
+		StringInfo command = makeStringInfo();
+		appendStringInfo(command, "SAVEPOINT savepoint_%u", subId);
+		RemoteSubTransactionCommand(connection, command);
+	}
```

Comment:
I'd like split this into Start/Finish* as well, to avoid adding latencies for all connections. I.e. in Start* send the `SAVEPOINT` to all asynchronously, and then only wait for all of them together.